tests/monotouch-test/CoreBluetooth/CentralManagerTest.cs

```diff
@@ -36,6 +36,10 @@ public class CBCentralManagerTest {
 		class ManagerDelegate : CBCentralManagerDelegate {
 			public AutoResetEvent Event { get; private set; }
 
+			public ManagerDelegate (IntPtr handle) : base (handle)
+			{
+			}
+
```

Comment:
That won't set the managed state (`Event`) back to a valid (non-null) value. It will likely work, just like rebuilding would have - i.e. your code change being irrelevant to the test success/failure.

My _guess_ is that the 5sec timeout occurred on the bot and, while `TearDown` happened the callback occurred, throwing before the timeout assertion could occur.

You can try to duplicate this locally with a much shorter timeout.

Reply:
Still can't repro...

I tried playing with `mgrDelegate.Event.WaitOne` and the test's timeout `[Timeout]` and couldn't get the `TearDown` to happen before the Asserts...

I'm gonna ignore the test, file a bug and merge that PR so we can get going.

Reply:
Mmh actually it might be `ScanForPeripherals` failing... not `RetrievePeripherals`